src/dag_combine.rs

```diff
@@ -0,0 +1,56 @@
+use digraph::PyDiGraph;
+use pyo3::prelude::*;
+use petgraph::algo;
+use pyo3::Python;
+use pyo3::types::PyAny;
+use std::cmp::Ordering;
+
+fn contains_weight(py: Python, dag: &PyDiGraph, obj: &PyObject) -> i32 {
+    let mut index :i32 = -1;
```

Comment:
Instead of using `-1` and `i32` you should probably use `Option<usize>` and `None` represents -1 like your using here. The reason I suggest this is because the max value for `i32` is less than the max size for `usize` so for very large graphs you could end up with an overflow issue here.